### PR TITLE
Implement self autoupdating BrightSign autorun.brs script

### PR DESCRIPTION
**Pull request by ilyai**
Related issue: https://github.com/OutOfAxis/pixelart-cms-client/issues/1716

### CODE REVIEW

File: autorun.brs [hunk 1]
```diff
@@ -265,21 +278,46 @@ Sub EnterEventLoop()
...
+          print "BS: Error saving screenshot"
+        end if
+        gaa.screenshotTimer.Start()
+      else if timerData = "checkUpdate" then
+        DebugLog("BS: Checking for update")
+        versionRequest = CreateObject("roUrlTransfer")
+        versionRequest.SetUrl("https://raw.githubusercontent.com/OutOfAxis/fugo-player-bs/main/latest.txt")
+        latestVersion = versionRequest.GetToString().Trim()
+        DebugLog("BS: Latest version: " + latestVersion)
+        if gaa.version = latestVersion then
```
Comment (mrsln) [inline autorun.brs]: you need to check if `latestVersion `'s `responseCode` equals to 200

File: autorun.brs [hunk 2]
```diff
@@ -265,21 +278,46 @@ Sub EnterEventLoop()
...
+          print "BS: Screenshot has been saved"
+        else
+          print "BS: Error saving screenshot"
+        end if
+        gaa.screenshotTimer.Start()
+      else if timerData = "checkUpdate" then
+        DebugLog("BS: Checking for update")
+        versionRequest = CreateObject("roUrlTransfer")
+        versionRequest.SetUrl("https://raw.githubusercontent.com/OutOfAxis/fugo-player-bs/main/latest.txt")
+        latestVersion = versionRequest.GetToString().Trim()
```
Comment (mrsln) [inline autorun.brs]: have you tested this without the internet connection?

File: autorun.brs [hunk 3]
```diff
@@ -275,8 +276,6 @@ Sub EnterEventLoop()
       else
         DebugLog("BS: Unknown eventData: " + type(eventData))
       endif
-    else if type(ev) = "roGpioButton" then
-      if ev.GetInt() = 12 then stop
```
Comment (mrsln) [inline autorun.brs:266]: wow, what did it do?

File: autorun.brs [hunk 4]
```diff
@@ -298,23 +297,39 @@ Sub EnterEventLoop()
...
-          responseCode = scriptRequest.GetToFile("autorun.tmp")
-          DebugLog("BS: Response code = " + stri(responseCode))
-          if responseCode = 200 then
-            DebugLog("BS: Performing update")
-            MoveFile("autorun.brs", "autorun.brs~")
-            MoveFile("autorun.tmp", "autorun.brs")
-            RebootSystem()
+        versionRequest.SetPort(versionRequestPort)
+        if versionRequest.AsyncGetToString() then
+          event = wait(5000, versionRequestPort)
```
Comment (mrsln) [inline autorun.brs:304]: waiting for 5s for the request to be finished? is this how it's done in Bright Script?

File: autorun.brs (hunk 3 above)
Comment (ilyai) [reply autorun.brs:266]: not sure, but it caused script error once network is disconnected

File: autorun.brs (hunk 4 above)
Comment (ilyai) [reply autorun.brs:304]: for how long it should wait?

Comment (mrsln) [reply autorun.brs:304]: no, just a weird syntax

**Review by mrsln — APPROVED** [review]

**Pull request merged**